controller/fleet_mock_test.go

```diff
@@ -37,6 +67,17 @@ func (fm *fleetMock) GetStatusWithExpression(exp *regexp.Regexp) ([]fleet.UnitSt
 	return args.Get(0).([]fleet.UnitStatus), args.Error(1)
 }
 func (fm *fleetMock) GetStatusWithMatcher(f func(string) bool) ([]fleet.UnitStatus, error) {
-	args := fm.Called(f)
-	return args.Get(0).([]fleet.UnitStatus), args.Error(1)
+	if fm.UseTestifyMock {
+		args := fm.Called(f)
+		return args.Get(0).([]fleet.UnitStatus), args.Error(1)
+	} else if fm.UseCustomMock {
+		fm.CustomMockUsed++
+		if fm.CustomMockUsed <= 3 {
+			return fm.FirstCustomMockStatus, nil
```

Comment:
I cannot make it work that way. In case you want to invest time here lets have a look at it together. Otherwise I vote for keeping it as it is right now.